bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/texteditor/FindReplaceOverlay.java

```diff
@@ -236,21 +236,57 @@ public void controlResized(ControlEvent e) {
 	private IPartListener partListener = new IPartListener() {
 		@Override
 		public void partActivated(IWorkbenchPart part) {
-			if (getShell() != null) {
-				getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
-			}
+			Display.getDefault().asyncExec(new Runnable() {
+				@Override
+				public void run() {
+					reactToListener();
+				}
+			});
 		}
 
 		@Override
 		public void partDeactivated(IWorkbenchPart part) {
-			// Do nothing
+			Display.getDefault().asyncExec(new Runnable() {
+				@Override
+				public void run() {
+					reactToListener();
+				}
+			});
 		}
 
 		@Override
 		public void partBroughtToTop(IWorkbenchPart part) {
-			if (getShell() != null) {
-				getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
+			Display.getDefault().asyncExec(new Runnable() {
+				@Override
+				public void run() {
+					reactToListener();
+				}
+			});
+		}
+
+		private void reactToListener() {
+			if (getShell() == null) {
+				return;
 			}
+
+			getShell().setVisible(isPartCurrentlyDisplayedInPartSash());
```

Comment:
The code called from `asyncExec()` **without** checking whether preconditions are still met : the `targetPart` is still opened and it's shell is non null.

Same for the code below in the `asyncExec()` call.

Reply:
I can add checks for whether the shell is still open, sure. How do I check whether the targetPart is still open? I didn't find a method for that

Reply:
targetPart.getSite().getPart() would be null.